metashare/api/validators.py

```diff
@@ -13,21 +13,23 @@ def process_field_name(self, field_name):
             return "name__iexact"
         return field_name
 
-    def filter_queryset(self, attrs, queryset):
+    def filter_queryset(self, attrs, queryset, serializer):
         """
         Filter the queryset to all instances matching the given attributes.
         """
+        # field names => field sources
+        sources = [serializer.fields[field_name].source for field_name in self.fields]
+
         # If this is an update, then any unprovided field should
-        # have its value set based on the existing instance attribute.
-        if self.instance is not None:
-            for field_name in self.fields:
-                if field_name not in attrs:
-                    attrs[field_name] = getattr(self.instance, field_name)
+        # have it's value set based on the existing instance attribute.
```

Comment:
```suggestion
        # have its value set based on the existing instance attribute.
```

Reply:
This was copied verbatim from the relevant bit of DRF source, so I should push that fix upstream too :)

Reply:
Maybe add a comment that this is copied from elsewhere?